libsmtutil/SolverInterface.h

```diff
@@ -213,7 +213,7 @@ class Expression
 		std::shared_ptr<ArraySort> arraySort = std::dynamic_pointer_cast<ArraySort>(_array.sort);
 		smtAssert(arraySort, "");
 		smtAssert(_index.sort, "");
-		smtAssert(*arraySort->domain == *_index.sort, "");
+		smtAssert(arraySort->domain->kind == _index.sort->kind, "");
```

Comment:
I think the same (or similar) should be done for the other cases as well. If we only check the `kind` we're bypassing a bunch of checks.

Reply:
Right, I commented only on that one, but was actually thinking about all the cases.